### PR TITLE
CORE-19451: Start managed key rotation from REST API

### PR DESCRIPTION
**Pull request by anien**

### CODE REVIEW

**Review by vkolomeyko — APPROVED** [review]
LGTM

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt [hunk 1]
```diff
@@ -235,8 +236,19 @@ class KeyRotationRestResourceImpl @Activate constructor(
...
+            checkNotNull(publishToKafka)
+        }
+
+        if (tenantId.isEmpty()) throw InvalidInputDataException(
+            "Cannot start key rotation. TenantId is not specified."
+        )
+
+        return doManagedKeyRotation(
+            tenantId,
+            publishRequests = { publishToKafka!!.publish(it) }
```
Comment (Sakpal) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt:250]: Minor: If `publishToKafka` is null, what error will we send back in API Call?

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt [hunk 2]
```diff
@@ -298,3 +310,20 @@ fun doKeyRotation(
     publishRequests(listOf(Record(REKEY_MESSAGE_TOPIC, requestId, keyRotationRequest, Instant.now().toEpochMilli())))
     return ResponseEntity.accepted(KeyRotationResponse(requestId, oldKeyAlias, newKeyAlias))
 }
+
+fun doManagedKeyRotation(
```
Comment (Sakpal) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt:314]: Minor: `private` if it is internally used?

Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt:314]: We want to use that function in test, hence can't be private. Originally, the code in this function was not separate, but included in `onNext` function. But we were unable to get it tested without including `lifecycle` and `OSGi`. This separation an option, how to make it testable.

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt (hunk 1 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt:250]: `checkNotNull` checks if `publishToKafka` is `null` and will throw an exception that is caught by `tryWithExceptionHandling`. This is used in multiple rest resource classes as `tryWithExceptionHandling` handles all exceptions and translate them to `HttpApiException`, which are usual error codes presented when dealing with HTTP requests. In this case, it throws `InternalServerException`, 500 error code. We verify this behaviour in our unit tests.

**Review by Sakpal — APPROVED** [review]
LGTM

**Review by simon-johnson-r3 — APPROVED** [review]

**Pull request merged**